cli/src/main.rs

```diff
@@ -0,0 +1,221 @@
+    };
+
+    // Initialize the REPL
+    let mut rl = DefaultEditor::new()?;
+
+    println!("Welcome to the SlateDB REPL!");
+    println!("WARNING: This is currently intended for development use.");
+    println!(
+        r#"
+  ____  _       _       ____  ____
+ / ___|| | __ _| |_ ___|  _ \| __ )
+ \___ \| |/ _` | __/ _ \ | | |  _ \
+  ___) | | (_| | ||  __/ |_| | |_) |
+ |____/|_|\__,_|\__\___|____/|____/
+"#
+    );
+    println!("Type 'help' to see available commands.\n");
+
+    loop {
+        // Prompt the user for input
+        let readline = rl.readline(">> ");
+        match readline {
+            Ok(line) => {
+                let trimmed = line.trim();
+                if trimmed.is_empty() {
+                    continue;
+                }
+                match rl.add_history_entry(trimmed) {
+                    Ok(_) => {}
+                    Err(e) => {
+                        eprintln!("Could not save line to history: {}", e)
+                    }
+                }
+
+                // Parse the input using shell-words to handle quotes
+                let parts_result = split(trimmed);
+                let parts = match parts_result {
+                    Ok(p) => p,
+                    Err(e) => {
+                        eprintln!("Error parsing input: {}", e);
+                        continue;
+                    }
+                };
+
+                if parts.is_empty() {
+                    continue;
+                }
+
+                let command = parts[0].to_lowercase();
+
+                match command.as_str() {
+                    "put" => {
+                        if parts.len() < 3 {
+                            println!("Usage: put <key> <value>");
+                            continue;
+                        }
+                        let key = &parts[1];
+                        let value = parts[2..].join(" "); // Support spaces in value
+                        match db_cli.put(key, &value).await {
+                            Ok(_) => println!("Successfully inserted key-value pair."),
+                            Err(e) => eprintln!("Error executing put: {}", e),
+                        }
+                    }
+                    "get" => {
+                        if parts.len() != 2 {
+                            println!("Usage: get <key>");
+                            continue;
+                        }
+                        let key = &parts[1];
+                        match db_cli.get(key).await {
+                            Ok(Some(value)) => println!("{}", value),
+                            Ok(None) => println!("Key not found."),
+                            Err(e) => eprintln!("Error executing get: {}", e),
+                        }
+                    }
+                    "delete" => {
+                        if parts.len() != 2 {
+                            println!("Usage: delete <key>");
+                            continue;
+                        }
```

Comment:
nitpick: similar code is repeated thrice, we can have a function like `validate_args(parts, valid_len, usage_message)` defined somewhere and then call it with in all the command cases

Reply:
I went ahead and tried to refactor things so that each command go its own implementation of a newly introduced `CliCommand` trait so that we can standardize this a little bit. This will also help with making the REPL and the CLI share more code when it gets to that.

LMKWYT